### ISSUE TITLE
<style> in rendered content from forks

### ISSUE THREAD

**Issue by mikicz** [mikicz/naucse.python.cz#2]
I put together a whitelist of allowed elements in content from forks - https://github.com/mikicz/naucse.python.cz/blob/arca_implementation/naucse/routes_util.py#L83, I basically extended the list of all elements used in naucse lessons, however there's one slight issue - I don't think having `<style>` there is a good idea, but the rendered jyputer notebooks contain them, so I had to put it on the whitelist for now.

`<style>` isn't in any of the other lessons and all the occurrences have the same content (see bellow), so it isn't even particularly important, the style could be moved to some css file and the elements themselves could be extracted when rendered from notebooks, but that's extra computing time. A different solution could be to check the content of `<style>` elements in `AllowedElementsParser` and only allow the styles from notebooks, but that solution isn't very pretty either...

Thoughts @hroncok @encukou?

The styles:
```
.dataframe thead tr:only-child th {
    text-align: right;
}

.dataframe thead th {
    text-align: left;
}

.dataframe tbody tr th {
    vertical-align: top;
}
```

**Comment by encukou** [comment]
Either of those two approaches should work. (And I guess that for the thesis, putting `<style>` on the whitelist would work as well, if you document it.)

Ideally, the Notebook formatter itself would have an option to disable inline `<style>`, for security purposes. Check if it's not already the case.

**Comment by mikicz** [comment]
Alright, so I check if the notebook formatter can disable styles, that's unfortunately not the case. The css we're talking about here is added in `HTMLFormatter.write_style` of Pandas (https://github.com/pandas-dev/pandas/blob/master/pandas/io/formats/format.py#L1158 at the time of writing) and it can be multiple different styles, however, all the styles are related to `.dataframe`. 

Since the styles actually seem relevant and we have to parse the entire rendered HTML anyway, I chose the second option on how to detect malicious css, throwing an exception if the css in `<style>` can't be parsed or if the styles aren't related to `.dataframe` (so even if Pandas changes some of the styles, no exception is thrown).

Fixed in: https://github.com/mikicz/naucse.python.cz/commit/df53b03660dfc892d89e4350b8a00e7f6698ec47
